src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -868,9 +869,38 @@ private void joinTokenRing(int delay) throws ConfigurationException
             setMode(Mode.WAITING_TO_BOOTSTRAP, "Awaiting start bootstrap call", true);
             bootstrapManager.awaitBootstrappable();
             if (SystemKeyspace.bootstrapInProgress())
-                logger.warn("Detected previous bootstrap failure; retrying");
+            {
```

Comment:
Could you write some kind of unit test for this? Even if it's (understandably!) hard to mock things end-to-end given the structure of the codebase you could still pull the new logic into another helper function and use the @VisibleForTesting annotation on it to get at this specifically